File: Application/src/main/java/com/shareyourproxy/api/domain/factory/ContactFactory.java
package com.shareyourproxy.api.domain.factory;

import com.shareyourproxy.api.domain.model.Contact;
import com.shareyourproxy.api.domain.realm.RealmString;

import java.util.HashSet;

import io.realm.RealmList;

/**
 * Factory for creating domain model {@link Contact}s.
 */
public class ContactFactory {

    /**
     * Convert a list of RealmString values into a HashSet of contact id Strings.
     *
     * @param values contact ids saved in realm
     * @return HashSet of contact id values
     */
    public static HashSet<String> getContactIdSet(RealmList<RealmString> values) {
        if (values != null) {
            HashSet<String> contactList = new HashSet<>(values.size());
            for (RealmString realmContact : values) {
                contactList.add(realmContact.getValue());
            }
            return contactList;
        }
        return null;
    }

}

File: Application/src/main/kotlin/com/shareyourproxy/api/domain/factory/ContactFactory.kt
package com.shareyourproxy.api.domain.factory

import com.shareyourproxy.api.domain.realm.RealmString
import io.realm.RealmList
import java.util.*

/**
 * Factory for creating domain model [Contact]s.
 */
internal object ContactFactory {
    /**
     * Convert a list of RealmString values into a HashSet of contact id Strings.
     * @param values contact ids saved in realm
     * @return HashSet of contact id values
     */
    fun getContactIdSet(values: RealmList<RealmString>): HashSet<String> {
        val contactList = HashSet<String>(values.size)
        values.forEach { contactList.add(it.value) }
        return contactList
    }
}
